frontend/packages/llama-stack-modular-ui/bff/internal/api/vector_db_handler_test.go

```diff
@@ -0,0 +1,389 @@
+		VectorDBID:     "test-db",
+		EmbeddingModel: "text-embedding-ada-002",
+	}
+
+	reqBody, err := json.Marshal(registrationRequest)
+	assert.NoError(t, err)
+
+	rr := httptest.NewRecorder()
+	req, err := http.NewRequest(http.MethodPost, "/api/v1/vector-dbs", bytes.NewBuffer(reqBody))
+	assert.NoError(t, err)
+	req.Header.Set("Content-Type", "application/json")
+
+	// Execute handler (no client in context)
+	app.RegisterVectorDBHandler(rr, req, nil)
+
+	// Should return 500 error
+	assert.Equal(t, http.StatusInternalServerError, rr.Code)
+
+	// Parse error response
+	body, err := io.ReadAll(rr.Body)
+	assert.NoError(t, err)
+
+	var errorResponse ErrorEnvelope
+	err = json.Unmarshal(body, &errorResponse)
+	assert.NoError(t, err)
+
+	assert.Equal(t, "500", errorResponse.Error.Code)
+	assert.Equal(t, "the server encountered a problem and could not process your request", errorResponse.Error.Message)
+}
+
+func TestRegisterVectorDBHandlerMissingVectorDBID(t *testing.T) {
+	app := createTestAppForVectorDB()
+
+	// Create request with missing vector_db_id
+	registrationRequest := VectorDBRegistrationRequest{
+		EmbeddingModel: "text-embedding-ada-002",
+		// VectorDBID missing
+	}
+
+	reqBody, err := json.Marshal(registrationRequest)
+	assert.NoError(t, err)
+
+	rr := httptest.NewRecorder()
+	req, err := http.NewRequest(http.MethodPost, "/api/v1/vector-dbs", bytes.NewBuffer(reqBody))
+	assert.NoError(t, err)
+	req.Header.Set("Content-Type", "application/json")
+
+	// Add HTTP client to context
+	httpClient := &integrations.HTTPClient{}
+	ctx := context.WithValue(req.Context(), constants.LlamaStackHttpClientKey, httpClient)
+	req = req.WithContext(ctx)
+
+	// Execute handler
+	app.RegisterVectorDBHandler(rr, req, nil)
+
+	// Should return 400 error
+	assert.Equal(t, http.StatusBadRequest, rr.Code)
+
+	// Parse error response
+	body, err := io.ReadAll(rr.Body)
+	assert.NoError(t, err)
+
+	var errorResponse ErrorEnvelope
+	err = json.Unmarshal(body, &errorResponse)
+	assert.NoError(t, err)
+
+	assert.Equal(t, "400", errorResponse.Error.Code)
+	assert.Contains(t, errorResponse.Error.Message, "vector_db_id is required")
+}
+
+func TestRegisterVectorDBHandlerMissingEmbeddingModel(t *testing.T) {
+	app := createTestAppForVectorDB()
+
+	// Create request with missing embedding_model
+	registrationRequest := VectorDBRegistrationRequest{
+		VectorDBID: "test-db",
+		// EmbeddingModel missing
+	}
+
+	reqBody, err := json.Marshal(registrationRequest)
+	assert.NoError(t, err)
+
+	rr := httptest.NewRecorder()
+	req, err := http.NewRequest(http.MethodPost, "/api/v1/vector-dbs", bytes.NewBuffer(reqBody))
+	assert.NoError(t, err)
+	req.Header.Set("Content-Type", "application/json")
+
+	// Add HTTP client to context
+	httpClient := &integrations.HTTPClient{}
+	ctx := context.WithValue(req.Context(), constants.LlamaStackHttpClientKey, httpClient)
+	req = req.WithContext(ctx)
+
+	// Execute handler
+	app.RegisterVectorDBHandler(rr, req, nil)
```

Comment:
It is fine for now - but in a follow up (just for easier readability and less verbosity) - 
1. We can have a setup code - which starts the BFF server with mock clients (from app.go)
2. Run all the handler tests one after the other.

This way we would not have to manually register every handler as it is being done now. It would be easier to add unit tests for various scenarios, instead of repeating the same setup again. 

This is where probably refactoring of using ginko/gomega would be helpful as they have the concept of `BeforeEach()` and `AfterEach()` (https://onsi.github.io/ginkgo/#extracting-common-setup-beforeeach) to do setup and cleanup.